samples/bluetooth/extended_adv/advertiser/src/main.c

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gap.h>
+
+static struct bt_conn *default_conn;
+
+enum bt_sample_adv_evt {
+	BT_SAMPLE_EVT_CONNECTED = BIT(0),
+	BT_SAMPLE_EVT_DISCONNECTED = BIT(1),
+};
+
+enum bt_sample_adv_st {
+	BT_SAMPLE_ST_ADV,
+	BT_SAMPLE_ST_CONNECTED,
+};
+
+static volatile enum bt_sample_adv_st app_st = BT_SAMPLE_ST_ADV;
+
+static struct k_poll_signal poll_sig = K_POLL_SIGNAL_INITIALIZER(poll_sig);
+static struct k_poll_event poll_evt = K_POLL_EVENT_INITIALIZER(K_POLL_TYPE_SIGNAL,
+						K_POLL_MODE_NOTIFY_ONLY, &poll_sig);
+
+static void connected_cb(struct bt_conn *conn, uint8_t err)
+{
+	printk("Connected (err 0x%02X)\n", err);
+
+	__ASSERT(conn == default_conn, "Unexpected connected callback");
+
+	if (err) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+
+	default_conn = bt_conn_ref(conn);
```

Comment:
Let's tie references to pointers. If `default_conn != NULL` (i.e. the variable holds a pointer), then it's wrong to overwrite `default_conn` without `bt_conn_unref(default_conn)`.

I believe you actually want to just drop this line.
```suggestion
```